config/unicorn.conf.rb

```diff
@@ -270,9 +291,14 @@ def master_sleep(sec)
 after_fork do |server, worker|
   DiscourseEvent.trigger(:web_fork_started)
   Discourse.after_fork
+  SignalTrapLogger.instance.after_fork
 
-  Signal.trap("USR2") { puts <<~MSG }
-      [#{Time.now.utc.strftime("%Y-%m-%dT%H:%M:%S.%6N")} ##{Process.pid}] Received USR2 signal, dumping backtrace for all threads
-      #{Thread.list.map { |t| "#{t.backtrace&.join("\n")}" }.join("\n\n")}
+  Signal.trap("USR2") do
+    message = <<~MSG
+    Unicorn worker received USR2 signal indicating it is about to timeout, dumping backtrace for main thread
+    #{Thread.current.backtrace&.join("\n")}
```

Comment:
I changed this to only dump the main thread. Unicorn workers don't timeout because of background threads so no point dumping the backtraces for those.